src/OrchardCore.Modules/OrchardCore.Taxonomies/Indexing/TaxonomyIndex.cs

```diff
@@ -35,21 +35,17 @@ public TaxonomyIndexProvider(IServiceProvider serviceProvider)
         public override void Describe(DescribeContext<ContentItem> context)
         {
             context.For<TaxonomyIndex>()
+                .When(contentItem => contentItem.IsPublished() || contentItem.Latest)
                 .Map(contentItem =>
                 {
-                    if (!contentItem.IsPublished())
```

Comment:
Cool for first pass.

I think on taxonomies we should add `Published` and `Latest` to the index.

because right now, I have a lot of SQL Queries, that do not `JOIN` on the `ContentItemIndex` because they did not need to. And I figure if I have them, other people do too, especially from the various issues where people have asked about how to query taxonomies with SQL Queries. 
So good to be able to keep to one query, no join. and easy to add `Published` and `Latest` with defaults for both (true, false) - I think I added this to `Alias` some time ago.

Reply:
> I think on taxonomies we should add Published and Latest to the index.

This is what it now does, with the above `.When` that i added

Reply:
Haha, go to bed. 
What I mean is we should track the `Published` and `Latest` values, in the Index itself

so the index is now
```
    public class TaxonomyIndex : MapIndex
    {
        public string TaxonomyContentItemId { get; set; }
        public string ContentItemId { get; set; }
        public string ContentType { get; set; }
        public string ContentPart { get; set; }
        public string ContentField { get; set; }
        public string TermContentItemId { get; set; }
    }
```

and we should make it

```
    public class TaxonomyIndex : MapIndex
    {
        public string TaxonomyContentItemId { get; set; }
        public string ContentItemId { get; set; }
        public string ContentType { get; set; }
        public string ContentPart { get; set; }
        public string ContentField { get; set; }
        public string TermContentItemId { get; set; }
        public bool Published { get; set; }
        public bool Latest { get; set; }
    }
```